### ISSUE TITLE
java.lang.AbstractMethodError on world load

### ISSUE THREAD

**Issue by kaikue** [MattDahEpic/MDECore1.8#17]
Full log: http://pastebin.com/9Z5Uzptv

Relevant part:
```
[19:17:48] [Server thread/ERROR] [FML]: Caught exception from MattDahEpic Core (mdecore)
java.lang.AbstractMethodError
	at com.mattdahepic.mdecore.command.AbstractCommand.registerCommandLogic(AbstractCommand.java:26) ~[mdecore-1.10.2-1.4.jar:?]
```

**Comment by MattDahEpic** [comment]
I am unable to reproduce this in a development environment, please try with [version 1.10.2-1.4.1](https://minecraft.curseforge.com/projects/mattdahepic-core-mdecore/files/2350441) and report the new crash.

**Comment by kaikue** [comment]
Same problem: http://pastebin.com/GyGE0hkY

My guess is that some mod adds a weird command that causes getCommandName() to throw an exception.

edit: I'm also using the latest Forge (12.18.2.2166) if that matters.

**Comment by MattDahEpic** [comment]
I've made a special version that logs all commands as they're being registered. Take a look at the log made with it and it should have a line that says something along the lines of `Registering command x.y.z.command` before the command that crashes.

https://dl.dropboxusercontent.com/u/146233520/mdecore-1.10.2-1.4.2-commanddebug.jar

**Comment by kaikue** [comment]
I did some more research (it's an Error not an Exception so it wasn't being caught, I had to recompile from source) and here are the erroring commands:

```
[17:36:18] [Server thread/ERROR] [mdecore]: Error initializing command com.mattdahepic.mdecore.command.logic.TPSLogic. Please report this to the mod author.
[17:36:18] [Server thread/ERROR] [mdecore]: Error initializing command com.mattdahepic.mdecore.command.logic.TPXLogic. Please report this to the mod author.
[17:36:18] [Server thread/ERROR] [mdecore]: Error initializing command com.mattdahepic.mdecore.command.logic.TPALogic. Please report this to the mod author.
```

This seems really weird, since those classes definitely do implement getCommandName(). It's also weird that this problem doesn't happen for you. But those commands are almost certainly the problem.

**Comment by kaikue** [comment]
Actually, I think there might be a conflict between `getCommandName()` in `AbstractSingleLogicCommand` and `getCommandName()` in `ICommandLogic`. Those three commands inherit both, and it might be getting confused, hence the error.

**Comment by MattDahEpic** [comment]
Since you already have an environment setup, would you mind testing this change?

**Comment by kaikue** [comment]
That still didn't work, but I made a pull request that does fix it: https://github.com/MattDahEpic/MDECore/pull/18

**Comment by kaikue** [comment]
If you're not going to change the command behavior, you might want to `return false` instead of throwing a `RuntimeException` when there's an error in `registerCommandLogic()` (so that it just skips the erroring command instead of completely crashing).

**Comment by MattDahEpic** [comment]
Requesting confirmation of bug fix on 1.10.2-1.5 with AODC.

**Comment by kaikue** [comment]
It works. Thanks!
